rules/no-complex-iteratee-expression.js

```diff
@@ -0,0 +1,115 @@
+'use strict';
+const {flatten} = require('lodash');
+const pluralize = require('pluralize');
+const avoidCapture = require('./utils/avoid-capture');
+const getDocumentationUrl = require('./utils/get-documentation-url');
+
+const MESSAGE_ID = 'no-complex-iteratee-expression';
+const messages = {
+	[MESSAGE_ID]: 'Move the complex iteratee expression out of the "for-of" header.'
+};
+
+const complexForSelector = [
+	'ForOfStatement',
+	`:not(${
+		[
+			'[right.type="Identifier"]',
+			'[right.type="MemberExpression"]',
+			'[right.type="CallExpression"][right.arguments.length=0]',
+			[
+				'[right.type="CallExpression"]',
+				'[right.callee.type="MemberExpression"]',
+				'[right.callee.object.type="Identifier"]',
+				'[right.callee.object.name="Object"]',
+				'[right.callee.property.type="Identifier"]',
+				'[right.callee.property.name=/(keys|values|entries)/]'
```

Comment:
```suggestion
				'[right.callee.property.name=/^(?:keys|values|entries)$/]'
```

Reply:
Need tests for it, previous version will ignore


```js
for (const foo of Object.xx_keys());
```

Reply:
Added in [`25b0af`](https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1131/commits/25b0af0ab70d5aab923f37ad77f4ff9ed7b5f544)